src/cljdoc/util/cache.clj

```diff
@@ -0,0 +1,123 @@
+(ns cljdoc.util.cache
+  "Provides `SQLCache`, an implementation of `clojure.core.cache/CacheProtocol`.
+  `SQLCache` makes following assumptions:
+
+  1. A table named CACHE exists in the database.
+  2. A column named KEY exists in CACHE.
+  3. A column named VALUE exists in CACHE.
+
+  It is used with `clojure.core.memoize/PluggableMemoization`
+  to provide a `clojure.core/memoize`  like API.
+
+  This namespace exposes `memo-sqlite` function which takes a
+  function to memoize and cache-spec, it retuns the memoized function.
+  "
+  (:require [clojure.core.cache :as cache]
+            [clojure.core.memoize :as memo]
+            [clojure.java.jdbc :as sql]
+            [taoensso.nippy :as nippy]))
+
+(cache/defcache SQLCache [state]
```

Comment:
You could make accessing of `cache-spec` fields easier by defining `SQLCache` like this:

```clojure
(cache/defcache SQLCache [db-spec key-prefix serialize-fn deserialize-fn])

(map->SQLCache {:db-spec ,,,
                :key-prefix "some-thing"
                :serialize-fn nippy/freeze
                :deserialize-fn nippy/thaw})
```

This way `serialize-fn` and so on could be accessed without having to retrieve it from the `state` map first.

> **Note:** I renamed `:db` to `:db-spec` in the code which matches more closely with the naming in cljdoc and `org.clojure/java.jdbc`.

Reply:
I changed the `:db` to `:db-spec`. After some implementing the `TTL` I had to do some refactoring now I need to pass the `cache-spec` to other functions so didn't change the field access part for now.